crates/nu-command/src/filesystem/glob.rs

```diff
@@ -186,6 +185,10 @@ impl Command for Glob {
             }
         };
 
+        let mut path = current_dir(engine_state, stack)?;
+        path.push(prefix);
+        let path = path.canonicalize()?;
```

Comment:
`current_dir()` should be guaranteed to return an absolute path (if it doesn't it's a bug). Therefore, canonicalize here is redundant, and also canonicalizing should be done via our `nu-path` crate. Using Rust's builtin canonicalize reads system environment, which is not supported in Nushell, we keep our own environment.